src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,239 @@
+import fs from 'fs'
+import * as simpleGit from 'simple-git'
+import {Writable} from 'stream'
+import {URL} from 'url'
+import {promisify} from 'util'
+
+import {renderGitError, renderProjectPathNotFoundWarning, renderSourcesNotFoundWarning} from './renderer'
+
+// NewSimpleGit returns a configured SimpleGit.
+export const newSimpleGit = (): simpleGit.SimpleGit => {
+  const options: simpleGit.SimpleGitOptions = {
+    baseDir: process.cwd(),
+    binary: 'git',
+    // We are invoking at most 3 git commands at the same time.
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// GitRemote returns the remote of the current repository.
+const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+  remotes.forEach((remote) => {
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
```

Comment:
Why do we need to do this?
Maybe a comment about this particularity would be useful.

Reply:
That was an arbitrary choice made when I wrote the RFC. The idea is that remotes don't really have any priority attached to them in git. So trying to pick the one called 'origin' since it's the default used by git before falling back to picking the first one seemed to make more sense to me.

I'll add a comment.

Reply:
Oh I see, but are you sure that when you return within a forEach, it's returned to the parent function?

Reply:
You're right, fixed the function and covered it with tests 🤦